pregen.js

```diff
@@ -91,3 +96,102 @@ export function preparePythonEnv(filePath, options) {
     }
   }
 }
+
+/**
+ * Method to check and prepare the environment for node
+ *
+ * @param {String} filePath Path
+ * @param {Object} options CLI Options
+ */
+export function prepareNodeEnv(filePath, options) {
+  // check tool for windows
+  for (const pt of options.projectType) {
+    const nodeVersion = pt.replace(/\D/g, "");
+    if (
+      pt.startsWith("nodejs") &&
+      nodeVersion &&
+      !process.env.NODE_INSTALL_ARGS
+    ) {
+      if (!isNvmAvailable()) {
+        if (process.env.NVM_DIR) {
+          // for scenarios where nvm is not present, but
+          // we have $NVM_DIR
+          // custom logic to find nvmNodePath
+          let nvmNodePath;
+          const possibleNodeDir = join(process.env.NVM_DIR, "versions", "node");
+          const nodeVersionArray = readdirSync(directoryPath, {
+            withFileTypes: true,
+          });
+          const nodeRe = new RegExp(`^v${nodeVersion}.`);
+          for (const nodeVersionsIter of nodeVersionArray) {
+            const fullPath = path.join(possibleNodeDir, nodeVersionsIter.name);
+            if (
+              nodeVersionsIter.isDirectory() &&
+              nodeRe.test(nodeVersionsIter.name)
+            ) {
+              nvmNodePath = path.join(fullPath, "bin");
+            }
+          }
+          if (nvmNodePath) {
+            doNpmInstall(filePath, nvmNodePath);
+          } else {
+            console.log(
+              `"node version ${nodeVersion} was not found. Please install it with 'nvm install ${nodeVersion}"`,
+            );
+            return;
+          }
+        } else {
+          console.log(
+            "Install nvm by following the instructions at https://github.com/nvm-sh/nvm",
+          );
+          return;
+        }
+      }
+      // set path instead of nvm use
+      const nvmNodePath = getNvmToolPath(nodeVersion);
+      doNpmInstall(filePath, nvmNodePath);
+    }
+  }
+}
+
+/**
+ * This method installs and create package-lock.json
+ *
+ * @param {String} filePath Path
+ * @param {String} nvmNodePath Path to node version in nvm
+ */
+export function doNpmInstall(filePath, nvmNodePath) {
+  // we do not install if INSTALL_ARGS set false
+  if (process.env.NODE_INSTALL_ARGS === false) {
+    return;
+  }
+
+  // const newPath =
+  // `${nvmNodePath}${delimiter}${process.env.PATH}`;
+
+  const resultNpmInstall = spawnSync(
+    process.env.SHELL || "bash",
+    [
+      "-i",
+      "-c",
+      `export PATH='${nvmNodePath}${delimiter}$PATH' && npm install --package-lock-only`,
```

Comment:
This won't work in CI such as GitHub actions. We have to construct these and pass via the `env` attribute in spawnSync.

Reply:
I tried to do it that way however that caused an issue, where the version is replaced by nvm when the shell starts.
e.g. if I add `v8.16.0` to path and run `spawnSync`, it is replaced by the default version of node `v22` in my case.